src/detect-engine-uint.c

```diff
@@ -211,9 +211,11 @@ DetectU32Data *DetectU32Parse (const char *u32str)
                 }
 
                 SCLogDebug("u32 is %"PRIu32" to %"PRIu32"", u32da.arg1, u32da.arg2);
-                if (u32da.arg1 >= u32da.arg2) {
-                    SCLogError(SC_ERR_INVALID_SIGNATURE, "Invalid u32 range. ");
-                    return NULL;
+                if (u32da.arg1 > u32da.arg2) {
+                    uint32_t temp = u32da.arg1;
+                    u32da.arg1 = u32da.arg2;
+                    u32da.arg2 = temp;
+                    SCLogWarning(SC_WARN_POOR_RULE, "Reversed u32 range. ");
```

Comment:
I think this is a clear error that we shouldn't accept and fix up, we should just invalidate the rule

Reply:
This was forced by unit test DetectICodeMatchTest01 :
```
    s = DetectEngineAppendSig(de_ctx, "alert icmp any any -> any any (icode:8<>20; sid:4;)");
    FAIL_IF_NULL(s);

    s = DetectEngineAppendSig(de_ctx, "alert icmp any any -> any any (icode:20<>8; sid:5;)");
    FAIL_IF_NULL(s);
```
Should we change this unit test then ?

Reply:
yeah lets not make the code bad because we have bad tests :)

Reply:
Ok will do.
I did not assume the test was bad ;-)